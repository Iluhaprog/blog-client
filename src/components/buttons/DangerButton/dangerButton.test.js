import React from 'react';
import { render, waitFor, fireEvent } from '@testing-library/react';
import { DangerButton } from '.';

test('Test view of danger button', () => {
    const { asFragment } = render(<DangerButton text='test'/>);
    expect(asFragment()).toMatchSnapshot();
});

test('Test onclick event of danger button', async () => {
    const handleClick = jest.fn();
    const { container } = render(<DangerButton text='test' onClick={handleClick}/>);
    fireEvent.click(container.querySelector('button'));
    await waitFor(() => {
        expect(handleClick).toHaveBeenCalledTimes(1);
    });
});