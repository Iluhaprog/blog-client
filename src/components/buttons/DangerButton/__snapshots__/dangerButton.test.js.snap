// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test view of danger button 1`] = `
<DocumentFragment>
  <button
    class="button danger-button"
  >
    test
  </button>
</DocumentFragment>
`;
